SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1664,24 +1664,28 @@ void USpatialReceiver::HandleRPC(const Worker_ComponentUpdateOp& Op)
 		return;
 	}
 
-	const TWeakObjectPtr<UObject> ActorReceivingRPC = PackageMap->GetObjectFromEntityId(Op.entity_id);
-	if (!ActorReceivingRPC.IsValid())
-	{
-		UE_LOG(LogSpatialReceiver, Error, TEXT("Entity receiving ring buffer RPC does not exist in PackageMap! Entity: %lld, Component: %d"), Op.entity_id, Op.update.component_id);
-		return;
-	}
-
 	// When migrating an Actor to another worker, we preemptively change the role to SimulatedProxy when updating authority intent.
-	// This causes the engine to print errors when we try and processed received RPCs while not authoritative. Instead, we early
-	// exit here, and the RPC will be processed by the server that receives authority.
+	// This can happen while this worker still has ServerEndpoint authority, and attempting to process a server RPC causes the engine
+	// to print errors if the role isn't Authority. Instead, we exit here, and the RPC will be processed by the server that receives
+	// authority.
 	const bool bIsServerRpc = Op.update.component_id == SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID;
-	const bool bActorRoleIsSimulatedProxy = Cast<AActor>(ActorReceivingRPC.Get())->Role == ROLE_SimulatedProxy;
-	if (bIsServerRpc && bActorRoleIsSimulatedProxy)
+	const bool bHaveServerEndpointAuthority = StaticComponentView->HasAuthority(Op.entity_id, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID);
```

Comment:
I believe that `StaticComponentView->HasAuthority` should be only be called if `bIsServerRpc` is true.